workflows/diagnostics/fv3net/diagnostics/offline/compute.py

```diff
@@ -230,16 +235,10 @@ def insert_prediction(ds: xr.Dataset, ds_pred: xr.Dataset) -> xr.Dataset:
     return xr.merge([safe.get_variables(ds, nonpredicted_vars), ds_target, ds_pred])
 
 
-def _get_predict_function(predictor, variables, grid):
+def _get_predict_function(predictor, variables):
     def transform(ds):
-        # Prioritize dataset's land_sea_mask if grid values disagree
-        ds = xr.merge(
-            [ds, grid], compat="override"  # type: ignore
-        )
-        derived_mapping = DerivedMapping(ds)
```

Comment:
Is the derived mapping `.dataset` called earlier such that derived inputs like cosine zenith angle are already in the dataset when it is used as the input for `.predict`?

Reply:
Yes, assuming that the `BatchesConfig` specified `needs_grid=True`, then it'll be done already on loading them.